File: tasks-3.4/3.4.f
from itertools import product

val = list(product([2, 3, 4, 5, 6, 7, 8, 9, 10, 'валет', 'дама', 'король', 'туз'], ['пик', 'треф', 'бубен', 'червей']))
s = input()
for i in val:
    if i[1] != s:
        print(i[0], i[1])
    else:
        continue

File: tasks-3.4/3.4.r
from itertools import product
expr = input()

print('a b c f')

for a, b, c in product((0, 1), repeat=3):
    f = eval(expr, {}, {'a': bool(a), 'b': bool(b), 'c': bool(c)})
    print(a, b, c, int(f))
